### PR TITLE
RELATED: BB-2328 SyntaxHighlightingInput callback to report cursor position

### PR DESCRIPTION
**Pull request by xMort**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7749/ : FAILURE in 3m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3883/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7750/ : FAILURE in 1m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3884/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7780/ : SUCCESS in 6m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3909/ : SUCCESS in 4s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7781/ : SUCCESS in 5m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3910/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7792/ : SUCCESS in 7m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3917/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7796/ : SUCCESS in 3m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3921/ : SUCCESS in 3s

File: src/components/simple/SyntaxHighlightingInput.tsx [hunk 1]
```diff
@@ -8,6 +8,8 @@ import "codemirror/addon/mode/simple";
 import "codemirror/addon/edit/matchbrackets";
 import "codemirror/addon/display/placeholder";
 
+const CODE_MIRROR_LINE_DELIMITER = "\n";
```
Comment (yanlipnican) [inline src/components/simple/SyntaxHighlightingInput.tsx:11]: I would rename it to `NEWLINE`, `END_OF_LINE`, `EOL` or something like that... It's not codemirror specific and it's not called line delimeter

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7818/ : SUCCESS in 4m 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3952/ : SUCCESS in 6s

File: src/components/simple/SyntaxHighlightingInput.tsx [hunk 2]
```diff
@@ -17,20 +19,57 @@ const defaultOptions = {
...
+        .split(CODE_MIRROR_LINE_DELIMITER)
+        .map(getLineLengthToCursor)
+        .reduce((sum, lineLength) => sum + lineLength, 0);
+};
+
 export const SyntaxHighlightingInput: React.FC<ISyntaxHighlightingProps> = props => {
-    const { value, onChange, formatting, customOptions, className } = props;
+    const { value, onChange, onCursor, formatting, customOptions, className } = props;
+
+    const reportCursorPosition = (_editor: CodeMirror.Editor): void => {
```
Comment (yanlipnican) [inline src/components/simple/SyntaxHighlightingInput.tsx:50]: `_editor` -> `editor`

File: src/components/simple/SyntaxHighlightingInput.tsx [hunk 3]
```diff
@@ -17,20 +19,57 @@ const defaultOptions = {
...
     const handleOnChange = (
         _editor: CodeMirror.Editor,
         _data: CodeMirror.EditorChange,
         value: string,
     ): void => {
         onChange(value);
+        reportCursorPosition(_editor);
+    };
+
+    const handleOnBlur = (_editor: CodeMirror.Editor): void => {
```
Comment (yanlipnican) [inline src/components/simple/SyntaxHighlightingInput.tsx:71]: `_editor` -> `editor`

**Comment by yanlipnican** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7820/ : SUCCESS in 2m 46s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3955/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4087/ : UNSTABLE in 11m 08s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/123/ : SUCCESS in 17s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7821/ : SUCCESS in 2m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1900/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2826/ : SUCCESS in 18m 43s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4090/ : UNSTABLE in 12m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/126/ : SUCCESS in 16s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7824/ : SUCCESS in 2m 45s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1903/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2830/ : SUCCESS in 8m 12s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4093/ : SUCCESS in 15m 46s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/129/ : SUCCESS in 15s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7828/ : SUCCESS in 5m 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1906/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2833/ : SUCCESS in 18m 33s

**Comment by xMort** [comment]
extended test - cafe

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3966/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3444/ : FAILURE in 10s

**Comment by xMort** [comment]
extended test

**Comment by yenkins** [comment]
Demo is running at https://client-demo.na.intgdc.com:50030/storybook/

**Comment by yenkins** [comment]
Demo is running at https://client-demo.na.intgdc.com:50042/storybook/

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3969/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3449/ : FAILURE in 6m 46s

**Pull request merged**